### ISSUE TITLE
MapArray of native types

### ISSUE THREAD

**Issue by Toldy** [Hearst-DD/ObjectMapper#175]
Hi,

I don't find how to map my JSON to an `Array<String>`
I tried with:
```
let awesomeArray = Mapper<String>().mapArray("[\"My\", \"pseudo\", \"is\", \"Toldino\", \"!\"]")
```
But obviously String doesn't extend from Mappable protocol !
My problem seems simple, no ?

Could you help me ? Thanks in advance..

**Comment by Toldy** [comment]
I looked your code and I understand why it doesn't work.
For a mapArray the JSON is convert 2 times:

`if let JSONArray = JSON as? [[String : AnyObject]] {`
`if let JSON = JSON as? [String : AnyObject] {`

So, it's obvious that my JSON cannot be cast in `Array<String>`.
Hope it helps !

**Comment by cezheng** [comment]
I guess you just need to use NSJSONSerialization to do the job. Using ObjectMapper is just overkill

**Comment by Toldy** [comment]
That is not a solution.. 
I am using ObjectMapper for ALL requests of my app and I would prefer not to use the NSJSONSerialization to keep my code well structured, portable.  

@cezheng In reality, my example is simpler than my problem. I didn't want to add more problems to the existing one 

But, if you need my whole problem to understand why it is not "overkill", this is it:

The JSON I want to parse:
` [{\"foo\":{\"type\":\"type_a\",\"id\":\"1\"}},\"+\",[\"12\",\"x\",{\"foo\":{\"type\":\"type_b\",\"id\":\"1\"}}]]" `

**Comment by Toldy** [comment]
That is not a solution.. 
I am using ObjectMapper for ALL requests of my app and I would prefer not to use the NSJSONSerialization to keep my code well structured, portable.  

@cezheng In reality, my example is simpler than my problem. I didn't want to add more problems to the existing one.

But, if you need my whole problem to understand why it is not "overkill", this is it:

The JSON I want to parse:
```json
[
   {
     "foo": {
        "type": "type_a",
        "id":"1"
      }
   },
   "+",
   [
      "12",
      "x",
      {
         "foo": {
            "type": "type_b",
            "id": "1"
         }
      }
   ]
]
```
So basically the syntax of the array is :
```
array-type → [ type ]
type → array-type | string-type | object-type
```

You should better understand my problem now;
I can't map the main array like `Mapper<AnyObject>().mapArray(JSONString)` because AnyObject doesn't inherit from `Mappable`

Regarding the code of ObjectMapper, my syntax can only be:
```
array-type → [ key: type ]
[...]
```
I can ask to my API devs to change the JSON, but that means a really big workload.

Thanks in advance

**Comment by cezheng** [comment]
I see. Then I guess the only way to work around this is to wrap the 

```
type → array-type | string-type | object-type
```
 into a custom class that implements Mappable
